File: ios/Samples/AndorraTelecomPublic/AndorraTelecomPublic/Helpers/LanguageHelper.swift
//
//  LanguageHelper.swift
//  AndorraTelecomPublic
//
//  Created by Francisco Mico on 07/08/2017.
//  Copyright © 2017 Liferay. All rights reserved.
//

import Foundation

class LanguageHelper {
    
    private var languageApp: String
    
    private static var sharedLanguageHelper: LanguageHelper = {
        let languageHelper = LanguageHelper()
        return languageHelper
    }()
    
    class func shared() -> LanguageHelper {
        return sharedLanguageHelper
    }
    
    init() {
        self.languageApp = NSLocale.current.languageCode!
    }
    
    func change(language: String) {
        let endIndex = language.index(language.startIndex, offsetBy: 2)
        self.languageApp = language.substring(to: endIndex)
    }
    
    func currentLanguage() -> String {
        return languageApp
    }
    
    let listLanguages = ["cat", "esp", "fra", "eng"]
    
    var formattedId: String {
        switch languageApp {
        case "ca":
            return "ca_ES"
        case "es":
            return "es_ES"
        case "fr":
            return "fr_FR"
        default:
            return "en_US"
        }
    }
    
    var threeLettersFormatted: String {
        switch languageApp {
        case "ca":
            return "cat"
        case "es":
            return "esp"
        case "fr":
            return "fra"
        default:
            return "eng"
        }
    }
    
    func url(page: Pages) -> String {
        let url = "https://www.andorratelecom.ad/c/portal/update_language?p_l_id=\(page.plid)&redirect=\(page.pathName)&languageId=\(self.formattedId)"
        print("\nURL: \(url)")
        return url
    }
    
    enum Pages: String {
        case index
        case mobileland29
        case roming
        case paquete69
        case optima
        case legal
        case map
        
        var plid: String {
            switch self {
            case .index:
                return "34684"
            case .mobileland29:
                return "100490"
            case .roming:
                return "100870"
            case .paquete69:
                return "108620"
            case .optima:
                return "106754"
            case .legal:
                return "34691"
            case .map:
                return "33998"
            }
        }
        
        var pathName: String {
            switch self {
            case .index:
                return "/particulars/inici/"
            case .mobileland29:
                return "/particulares/som-telefonia-movil/forfets-mobiland/forfet-mobiland-29/"
            case .roming:
                return "/particulares/som-telefonia-movil/otros-forfets/forfet-roaming-espana/"
            case .paquete69:
                return "/particulares/som-internet-fija/paquetes-internet-fija/paquet-69/"
            case .optima:
                return "/particulares/som-internet-fija/paquetes-internet-fija/optima/"
            case .legal:
                return "/avis-legal/"
            case .map:
                return "/web/comercial/on-som"
            }
        }
    }
    
}

extension String {
    func localized() -> String? {
        if let path = Bundle.main.path(forResource: LanguageHelper.shared().currentLanguage(), ofType: "lproj") {
            if let bundle = Bundle(path: path) {
                return NSLocalizedString(self, tableName: "translations", bundle: bundle, value: "", comment: "")
            }
        }
        
        return nil;
    }
}

File: xamarin/Samples/AndorraTelecom-Android/Util/LanguageHelper.cs
using System;

namespace AndorraTelecomiOS.Util
{
    public static class LanguageHelper
    {
        public static string[] ListLanguages = { "esp", "cat", "fra", "eng" };

        public static string CurrentLanguage = "en_US";

        public static string LanguageApp {
            get
            {
                var Language = Java.Util.Locale.Default.Language;
                switch (Language)
                {
                    case "es":
                        return "es_ES";
                    case "ca":
                        return "ca_ES";
                    case "fr":
                        return "fr_FR";
                    default:
                        return "en_US";
                }
            }
            set
            {
                CurrentLanguage = value;
            }
        }

        public static int CurrentSelectedLanguageInDialog
        {
            get
            {
                switch (CurrentLanguage)
                {
                    case "es_ES":
                        return 0;
                    case "ca_ES":
                        return 1;
                    case "fr_FR":
                        return 2;
                    default:
                        return 3;
                }
            }
        }

        public enum Pages { Index, Mobile, Roaming, Paquete69, Optima, Legal, Map };

        /* Methods */

        public static string GetPlid(Pages Page)
        {
            switch (Page)
            {
                case Pages.Mobile:
                    return "839763";
                case Pages.Roaming:
                    return "841024";
                case Pages.Paquete69:
                    return "108620";
                case Pages.Optima:
                    return "106754";
                case Pages.Legal:
                    return "34691";
                case Pages.Map:
                    return "33998";
                default:
                    return "34684";
            }
        }

        public static string GetPathName(Pages Page)
        {
            switch (Page)
            {
                case Pages.Mobile:
                    return "/particulars/telefonia-mobil/forfets-mobiland/contracte/";
                case Pages.Roaming:
                    return "/particulars/telefonia-mobil/serveis-complementaris/forfet-roaming/";
                case Pages.Paquete69:
                    return "/particulars/internet-fixa/internet-i-telefonia-fixa/paquet-69/";
                case Pages.Optima:
                    return "/particulars/internet-fixa/internet-i-telefonia-fixa/optima/";
                case Pages.Legal:
                    return "/avis-legal/";
                case Pages.Map:
                    return "/web/comercial/on-som";
                default:
                    return "/particulars/inici/";
            }
        }

        public static string Url(Pages Page)
        {
            var url = "https://www.andorratelecom.ad/c/portal/update_language?p_l_id=" 
                + GetPlid(Page) + "&redirect="
                + GetPathName(Page) + "&languageId=" 
                + CurrentLanguage + "";
            
            Console.WriteLine("URL: " + url);

            return url;
        }

        public static void SetCurrentLanguage(int SelectedLanguage)
        {
            switch(SelectedLanguage)
            {
                case 0:
                    LanguageApp = "es_ES";
                    break;
                case 1:
                    LanguageApp = "ca_ES";
                    break;
                case 2:
                    LanguageApp = "fr_FR";
                    break;
                default:
                    LanguageApp = "es_US";
                    break;
            }
        }
    }
}